clients/gtest/rocblas_gtest_main.cpp

```diff
@@ -159,46 +159,70 @@ static void rocblas_set_listener()
 /*********************************************
  * Signal-handling for detecting test faults *
  *********************************************/
-// Id of the thread which is catching signals
-static volatile pthread_t rocblas_test_sighandler_tid;
 
-// sigjmp_buf describing stack frame to go back to
-static sigjmp_buf rocblas_sigjmp_buf;
+// State of the signal handler
+static struct
+{
+    // Whether rocblas_sigjmp_buf is set and catching signals is enabled
+    volatile sig_atomic_t rocblas_sighandler_enabled;
+
+    // Id of the thread which is catching signals
+    volatile pthread_t rocblas_test_sighandler_tid;
+
+    // sigjmp_buf describing stack frame to go back to
+    sigjmp_buf rocblas_sigjmp_buf;
+} state;
```

Comment:
The structure containing the state of the signal handler. It is pushed/popped when nested.